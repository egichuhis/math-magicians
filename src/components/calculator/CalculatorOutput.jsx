import React from 'react';
import PropTypes from 'prop-types';

const CalculatorOutput = ({ calculatorData }) => (
  <div className="output" id="output">{calculatorData.next || calculatorData.total || '0'}</div>
);

CalculatorOutput.propTypes = {
  calculatorData: PropTypes.shape({
    total: PropTypes.string,
    next: PropTypes.string,
    operation: PropTypes.string,
  }),
};

CalculatorOutput.defaultProps = {
  calculatorData: {
    total: '',
    next: '',
    operation: '',
  },
};

export default CalculatorOutput;
