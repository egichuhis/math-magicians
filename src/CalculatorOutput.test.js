import React from 'react';
import { render } from '@testing-library/react';
import CalculatorOutput from './components/calculator/CalculatorOutput';

test('CalculatorOutput renders correctly', () => {
  const { container } = render(<CalculatorOutput />);
  expect(container).toMatchSnapshot();
});
