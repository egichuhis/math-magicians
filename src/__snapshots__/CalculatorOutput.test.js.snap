// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorOutput renders correctly 1`] = `
<div>
  <div
    class="output"
    id="output"
  >
    0
  </div>
</div>
`;
